tt_metal/llrt/hal.hpp

```diff
@@ -41,7 +41,8 @@ enum class HalMemAddrType : uint8_t {
     CORE_INFO = 7,
     GO_MSG = 8,
     LAUNCH_MSG_BUFFER_RD_PTR = 9,
-    COUNT = 10
+    DRAM_BARRIER = 10,
```

Comment:
The `HalMemAddrType`s really are `HalSramAddrType` as they are used to build memory map of SRAM of the programmable core types. `DRAM_BARRIER` is a reserved address in DRAM. 

I think this should be removed from `HalMemAddrType` and be tracked separately. Similarly, we shouldn't be adding this addr to the `mem_map_bases` or `mem_map_sizes` because those are also just for sram of the programmable core.

Reply:
Hmmm…

So do we need a new Hal interface/function then?

get_dev_addr requires a processor core type as the first argument.

Like get_addr(enum type)

We could also bypass a getter and have a public class member…

tt::tt_metal::Hal::DRAM_BARRIER_BASE

Reply:
I would avoid making `DRAM_BARRIER_BASE` a public class member because in the future we could want to pull out some DRAM specific config. 

I think we could overload `get_dev_addr(HalMemType, AddrType)`. The thing with this is that the L1 addresses (what is currently enumerated in `HalMemAddrType`) don't make sense to be requested without a core type. We could avoid this by design if we introduce another enum for addresses like the DRAM barrier base that do not depend on core type but this makes me think we could just rename `HalMemAddrType` to `HalL1AddrType` and then add a `HalDramAddrType` and have `get_dram_dev_addr` 

thoughts @pgkeller ?

Reply:
Renaming HalMemAddrType to HalL1AddrType and adding HalDramAddrType makes sense to me.  We could just overload get_dev_addr w/ the memory type.

Reply:
@pgkeller @abhullar-tt 

I applied these changes in the most recent push.